pastorsj/pandora-clone
# User Services

## Acceptance Criteria
* Three new routes for users, one is protected
* Added ability to register a user
* Added ability to login
* Added ability to retrieve user information given id

## How to test
* In your Neo4j database console (http://localhost:7474/browser/), run the following command
```
create constraint on (u:User) assert u.username is unique;
```
* Confirm that you can register at the /register route. In the body, send
```json
{
  "username": "your_username",
  "password": "your_password",
  "email": "your_email"
}
```
You should receive a [JWT](https://jwt.io) as a result.
* Confirm that you can login with your username and password. You should receive a JWT as a return.
* Run the register command again and confirm that you can't have the same username in the database.
* Confirm that at the route /user/:id, a get request with a user id will be rejected unless you send the request with the JWT in the Authorization header. Make sure to added Bearer to the front of the token with a space separating them.
* Confirm that there are no regressions.
